### PR TITLE
 From Leonard ???

### PR DESCRIPTION
**Pull request by LeonardLT**
fixed  userApi.js and register.jsx.
specs is not finished
need some specs to test the userApi.

### CODE REVIEW

**Comment by xiayu0321** [comment]
:+1:

File: public/component/register.jsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ export default class Register extends Component {
     }
 
     render() {
-        return <div>
+        return <form onSubmit={this.onSumbit.bind(this)}>
```
Comment (freewind) [inline public/component/register.jsx:16]: _onSumbit

File: public/component/register.jsx [hunk 2]
```diff
@@ -51,10 +51,10 @@ export default class Register extends Component {
...
                 </div>
-                <input type="submit" value="注册" className="btn btn-primary" onClick={this.onSumbit.bind(this)}/>
+                <input type="submit" value="注册" className="btn btn-primary"/>
                 <span>有账号?<a className="to_register">登陆 </a></span>
             </div>
-        </div>
+        </form>
     }
 
     onHandlerName(event) {
```
Comment (freewind) [inline public/component/register.jsx:60]: rename?

_onNameChanged

File: public/component/register.jsx [hunk 3]
```diff
@@ -83,33 +83,40 @@ export default class Register extends Component {
...
+            alert('confirm-password and password is not the same one!');
+        }
+        else {
+            request.post('/api/user')
+                .send({
+                    name: this.state.name,
+                    email: this.state.email,
+                    phone: this.state.phone,
+                    password: this.state.password,
+                    confirmPassword: this.state.confirmPassword
```
Comment (freewind) [inline public/component/register.jsx:103]: confirmPassword ???

File: public/component/register.jsx [hunk 4]
```diff
@@ -83,33 +83,40 @@ export default class Register extends Component {
...
+                })
+                .end((err, res) => {
+                    if (err) {
+                        alert(res.statusCode + ',' + res.text);
```
Comment (freewind) [inline public/component/register.jsx:107]: res can be null

File: public/component/register.jsx [hunk 5]
```diff
@@ -83,33 +83,40 @@ export default class Register extends Component {
...
+                        return console.error(err + ',' + res.text)
+                    }
+                    console.log(res.statusCode);
+                    if (res.statusCode == 201) {
+                        alert(res.statusCode + ",register success!");
+                        self.location = '/#/indexRent'
```
Comment (freewind) [inline public/component/register.jsx:113]: hashHistory.push('/indexRent')

Comment (freewind) [inline public/component/register.jsx:113]: index-rent

File: public/component/register.jsx [hunk 6]
```diff
@@ -83,33 +83,40 @@ export default class Register extends Component {
...
+                    confirmPassword: this.state.confirmPassword
+                })
+                .end((err, res) => {
+                    if (err) {
+                        alert(res.statusCode + ',' + res.text);
+                        return console.error(err + ',' + res.text)
+                    }
+                    console.log(res.statusCode);
+                    if (res.statusCode == 201) {
+                        alert(res.statusCode + ",register success!");
```
Comment (freewind) [inline public/component/register.jsx:112]: ???

File: public/component/register.jsx [hunk 7]
```diff
@@ -83,33 +83,40 @@ export default class Register extends Component {
...
+                    name: this.state.name,
+                    email: this.state.email,
+                    phone: this.state.phone,
+                    password: this.state.password,
+                    confirmPassword: this.state.confirmPassword
+                })
+                .end((err, res) => {
+                    if (err) {
+                        alert(res.statusCode + ',' + res.text);
+                        return console.error(err + ',' + res.text)
```
Comment (freewind) [inline public/component/register.jsx:108]: 分别处理不同的code

**Pull request merged**